compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -526,7 +563,9 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
                             .emit()
                     }
                 }
-            })
+            };
+
+            guar = Some(e)
```

Comment:
What's this for?

Reply:
Adding the `generic_param_scope` argument to a function call above changed formatting a bit which combined with it all being wrapped in a `Some` resulted in like 100 lines getting reformatted and indented an extra tab. I can revert this to how it used to be if you'd prefer to have the extra indent

Reply:
personally don't care about extra indents. github has "ignore whitespace" for review purposes, but up to you. I just didn't see where this statement was coming from, but maybe I was not reading correctly xd

Reply:
Oh yeah it was pre-existing. The code used to be `guar = Some(if let Some(guar) = ....` and i turned it into `let e = if let Some(guar) = ...` with the `guar = Some9e)`